### PR TITLE
Ben/auto version

### PR DESCRIPTION
**Pull request by benzcash**
Added a few more build details.

### CODE REVIEW

File: cmd/root.go [hunk 1]
```diff
@@ -182,7 +182,7 @@ func startServer(opts *common.Options) error {
 		os.Exit(1)
 	}()
 
-	common.Log.Infof("Starting gRPC server on %s", opts.BindAddr)
+	common.Log.Infof("Starting gRPC server version %s on %s", common.Version, opts.BindAddr)
```
Comment (LarryRuane) [inline cmd/root.go:185]: I think it would be good to log your improved version information here too, at startup.

Comment (benzcash) [reply cmd/root.go:185]: Added!

**Comment by benzcash** [comment]
> As long as we're printing all this, what do you think about adding the tree hash

Honestly, I'm not familiar with it.

I'm just used to using the output from `git rev-parse HEAD` to point to the last entry in `git log`.

I'll take a look at how to use it, I understand those advantages!

**Comment by LarryRuane** [comment]
> I'm just used to using the output from `git rev-parse HEAD` to point to the last entry in `git log`.

Yes, that prints the latest commit hash, which is good, let's keep that; tree-hash would just be in addition (no reason not to have both).

**Comment by LarryRuane** [comment]
Ben, I just thought of one more thing, we should change this line: https://github.com/zcash/lightwalletd/blob/ben/auto-version/frontend/service.go#L191
Would you mind doing that? I don't think that string needs to include all the detailed stuff, just the `common.Version` string.

**Comment by benzcash** [comment]
> Would you mind doing that?

Done!

**Pull request merged**